File: learn-algo/leetcode/506/go/506.go
package main

import (
	"fmt"
	"sort"
	"strconv"
)

var t []string = []string{"Gold Medal", "Silver Medal", "Bronze Medal"}

func findRelativeRanks(score []int) []string {
	type pair struct {
		index int
		value int
	}

	var sli []pair

	for i, v := range score {
		sli = append(sli, pair{i, v})
	}

	sort.Slice(sli, func(a, b int) bool {
		return sli[a].value > sli[b].value
	})

	var res = make([]string, len(score))

	for i, _ := range sli {
		if i < 3 {
			res[sli[i].index] = t[i]
		} else {
			res[sli[i].index] = strconv.Itoa(i + 1)
		}
	}

	return res
}

func main() {
	var in = []int{5, 4, 3, 2, 1}
	res := findRelativeRanks(in)
	fmt.Println(res)
}

File: learn/learn-algo/leetcode/506/c++/506.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#include "../../tools/tools-lib/tools.h"

using namespace std;

class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        int n = score.size();
        vector<pair<int,int>> vec;
        for(int i = 0; i < n; ++i) {
            pair<int, int> t = pair<int,int>(score[i], i);
            vec.push_back(t);
        }

        sort(vec.begin(), vec.end(), [](auto a, auto b){return a.first > b.first;});

        vector<string> t = {"Gold Medal","Silver Medal","Bronze Medal"};

        vector<string> res(n);
        for(int i = 0; i < score.size(); ++i) {
            if(i < 3) {
                res[vec[i].second] = t[i];
            } else {
                res[vec[i].second] = to_string(i);
            }

        }

        return res;
    }
};

int main() {
    Solution s;
    vector<int> in = {10,3,8,9,4};
    auto res = s.findRelativeRanks(in);
    printC(res);
    return 0;
}